pkg/routes/routes.go

```diff
@@ -141,42 +140,48 @@ func SetRoutes(r *gin.Engine) {
 		c.JSON(http.StatusOK, elastic.NewESXPack(c))
 	})
 
-	r.POST("/es/_search", AuthMiddleware, ESMiddleware, search.SearchDSL)
-	r.POST("/es/_msearch", AuthMiddleware, ESMiddleware, search.MultipleSearch)
-	r.POST("/es/:target/_search", AuthMiddleware, ESMiddleware, search.SearchDSL)
-	r.POST("/es/:target/_msearch", AuthMiddleware, ESMiddleware, search.MultipleSearch)
+	r.POST("/es/_search", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, search.SearchDSL)
+	r.POST("/es/_msearch", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, search.MultipleSearch)
+	r.POST("/es/:target/_search", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, search.SearchDSL)
+	r.POST("/es/:target/_msearch", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, search.MultipleSearch)
 
 	r.GET("/es/_index_template", AuthMiddleware, ESMiddleware, index.ListTemplate)
 	r.POST("/es/_index_template", AuthMiddleware, ESMiddleware, index.CreateTemplate)
-	r.PUT("/es/_index_template/:target", AuthMiddleware, ESMiddleware, index.CreateTemplate)
-	r.GET("/es/_index_template/:target", AuthMiddleware, ESMiddleware, index.GetTemplate)
-	r.HEAD("/es/_index_template/:target", AuthMiddleware, ESMiddleware, index.GetTemplate)
-	r.DELETE("/es/_index_template/:target", AuthMiddleware, ESMiddleware, index.DeleteTemplate)
+	r.PUT("/es/_index_template/:target", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, index.CreateTemplate)
+	r.GET("/es/_index_template/:target", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, index.GetTemplate)
+	r.HEAD("/es/_index_template/:target", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, index.GetTemplate)
+	r.DELETE("/es/_index_template/:target", AuthMiddleware, ESMiddleware, IndexAliasMiddleware, index.DeleteTemplate)
```

Comment:
We are not use alias every where, maybe just for search

Reply:
fixed